inc/Engine/Media/AboveTheFold/AJAX/Controller.php

```diff
@@ -248,4 +253,63 @@ public function check_lcp_data() {
 
 		wp_send_json_error( 'data does not exist' );
 	}
+
+	/**
+	 * Validate image object.
+	 *
+	 * @param object $image_object Image full object.
+	 * @return bool
+	 */
+	private function validate_image( $image_object ) {
+		$valid_image = ! empty( $image_object->src ) ? $this->validate_image_src( $image_object->src ?? '' ) : true;
+
+		/**
+		 * Filters If the image src is a valid image or not.
+		 *
+		 * @param bool   $valid_image Valid image or not.
+		 * @param string $image_src_url Image src url.
+		 * @param object $image_object Image object with full details.
+		 */
+		return (bool) apply_filters( 'rocket_atf_valid_image', $valid_image, $image_object->src, $image_object );
+	}
+
+	/**
+	 * Make sure that this url is valid image without loading the image itself.
+	 *
+	 * @param string $image_src Image src url.
+	 * @return bool
+	 */
+	private function validate_image_src( $image_src ) {
+		if ( empty( $image_src ) ) {
+			return false;
+		}
+
+		if ( preg_match( '/^chrome-[^:]+:\/\/[a-zA-Z0-9\-_]+\/?/', $image_src ) ) {
+			return false;
+		}
```

Comment:
That's a good idea to bailout early when finding this chrome scheme but what do u think if we parsed the url to get the scheme also and we can exclude whatever scheme we want, because we may increase this list of exclusions in the future, so what do u think if we use `wp_parse_url` to get also the scheme and exclude the scheme `chrome-extension`
So if we wanted to increase the list we can make it an array of exclusions and have a filter.